src/hotspot/share/logging/logDecorators.hpp

```diff
@@ -99,6 +127,20 @@ class LogDecorators {
     return _name[decorator][1];
   }
 
+  template<typename... Decorators>
+  static uint mask_from_decorators(LogDecorators::Decorator first, Decorators... rest) {
+    uint bitmask = 0;
+    LogDecorators::Decorator decorators[1 + sizeof...(rest)] = { first, rest... };
+    for (const LogDecorators::Decorator decorator : decorators) {
+      bitmask |= mask(decorator);
+    }
+    return bitmask;
+  }
+
+  // Check if we have some default decorators for a given LogSelection. If that is the case,
+  // the output parameter mask will contain the defaults-specified decorators mask
+  static bool has_disabled_default_decorators(const LogSelection& selection, const DefaultUndecoratedSelection* defaults = default_decorators, size_t defaults_count = number_of_default_decorators);
```

Comment:
I was trying to think if we could make this mockable without the incomplete object types (`const DefaultUndecoratedSelection* defaults = default_decorators, size_t defaults_count = number_of_default_decorators`).  

Maybe have the mockable part private (we already friend the gtest) and only have a `static bool has_disabled_default_decorators(const LogSelection& selection)` public (which calls this on the inside). 

But I am fine with this as it currently is.